index.js

```diff
@@ -0,0 +1,217 @@
+	// the search input will show him the different airports linked to his input.
+	// It is a search method linked to a JSON file that holds the IATA code and the name
+	// of the airports we need.
+	async options(fieldId, search) {
+		// const API_BASE_URL = `https://skyscanner-skyscanner-flight-search-v1.p.rapidapi.com/apiservices`;
+		if (airports) {
+			logger.info("Sending preloaded airports");
+			return this.getIATA(airports, search);
+		} else if (fs.existsSync('./airports.json')) {
+			logger.info('Loading places from a file.');
+			const airports = require('./airports.json');
+			return this.getIATA(airports, search);
+		}
+	}
+
+	// Parameters needed to create the search method and collect the right datas: IATAs and names of
+	// the airports.
+	/**
+	* @param {*} airports
+	* @param {String} search 
+	*/
+	async getIATA(airports, search) {
+		if (search != null) {
+			search = search.trim().toLowerCase();
+		}
+		let option = [];
+		for (const airport of airports) {
+			// Datas needed: the user will enter the name of the airport (value) which is
+			// linked to the IATA (key).
+			// The IATA is needed cause it will be added in the API url to allow the
+			// collect of the price.
+			const key = airport.iata;
+			let value = airport.name || '';
+			if (!search || value.toLowerCase().includes(search)) {
+				option.push({
+					key: key,
+					value: value
+				});
+			}	
+		}
+		return option;
+	}	
+
+	// Store price obtained from first loaded.
+	// The price is updated every 24 hours.
+	// This price is compared with the price collected from the API every minute.
+	storeFirstPriceOfTheDay(price) {
+		const firstPriceDate = this.store.get("firstPriceDate");
+		if(firstPriceDate == null || new Date().getTime() - firstPriceDate > 24*3600*1000) {
+			this.store.put("firstPrice", price);
+			this.store.put("firstPriceDate", new Date().getTime());
+		}
+		return true;
+	}
+
+	// Retrieve stored price data.
+	getFirstPriceOfTheDay() {
+		return this.store.get("firstPrice");
+	}
+
+	async run() {
+		// Compare the current price to the first price
+		return this.getPrice().then(new_price => {
+			this.storeFirstPriceOfTheDay(new_price);
+			const first_price = this.getFirstPriceOfTheDay();
+			logger.info(`The new price is ${new_price}`);
+			logger.info(`The first price is ${first_price}`);
+			logger.info(`The threshold is ${this.config.maxAffordablePrice}`);
+			let color;
+			let message;
+			// If there is no flight listed.
+			if (new_price == null) {
+				color = '#DF0101'; // red
+				message = `This flight is not listed. Please modify your request.`;
+			}
+			else if (new_price <= this.config.maxAffordablePrice) {
+				if (first_price == null) {
+					color = '#FFFF00'; // yellow
+					message = `The best price for this flight: from ${this.config.departurePlace}
+					to ${this.config.destinationPlace} the ${this.config.departureDate} is 
+					${new_price} ${this.config.currency}.`;
```

Comment:
Sounds weird... 
Also you can use `title` of Signal if you want to put less content in the `message` attribute itself.
Can you please ask advices to @supercobra for the content of the message for each use case